Updated reference implementation

zzyzek opened the pull request:
Reference implementation (in JS) with 2d/3d async and synchronous d2xy[z]/xy[z]2d functions.

We need to discuss the implementation a bit, which we can continue in issue #4, but now we have a starting point along with some tests to give us confidence of it working.

zzyzek committed:
reference implementation of gilbert++ algorithm

* has rudimentary tests that enumerate up to 13x13x13
  - ignore some known band conditions
  - need to still put in some simple tests to test "negative" volume

zzyzek committed:
Merge remote-tracking branch 'upstream/main'

zzyzek committed:
wip

* adding more to the conclusion (some verbiage from roadmap issue #2)
* adding 'hook' issue to a discussion subsection in the algorithm section
  (suggestion from issue #4)
* some notes that may go into the related work section about matrix
  multiplaction on GPU using Hilbert curve for locality/caching optimization

zzyzek committed:
"dynamic" gilbert++ algorithms

* change endpoint location to always have a Hamiltonian path, 2d and 3d functions

zzyzek committed:
wip

* adding 3d gilbert curve examples
  - still fussing with which should be representative
  - decided on full width figure
* adding a 'force width' option to the gilbert++ helper functions

zzyzek committed:
local symlink

zzyzek committed:
comments

* the curve generation got complex so I'm being kind to my future self
  and commenting the motivation and giving a high level description

zzyzek commented:
* Starting work on example 3d
* Updated reference implementation with helper functions (dynamic endpoint choice)

zzyzek committed:
wip

* edits
* adding 2D d2xyz and xyz2 algorithms in appendix
* adding inBounds function to appendix

zzyzek committed:
3d gilbert examples

zzyzek committed:
wip

* start of gilbert3d figures
  - position is pretty bad, maybe it'll get better with more text but as is,
    the eccentric and example 3d pictures take up their own pages
  - still need caption for examples

zzyzek commented:
* added 3d example figures
* other editing

zzyzek committed:
wip

* editing

jakubcerveny commented:
Love the 3D figures!

zzyzek committed:
changed function names to reflect adaptive

* `Gilbert3DAdapt_d2xyz` etc.
* comment cruft cleanup

zzyzek committed:
wip

* writing, editing, etc.

zzyzek committed:
spelling

zzyzek committed:
writing

* wip

zzyzek committed:
writing

* subsections to algorithms on extensions and pitfalls
* editing intro and algorithms

zzyzek committed:
wip

zzyzek committed:
eccentric appendix calculations

zzyzek committed:
wip

* editing, updates

zzyzek committed:
gilbert3dpp.js

* exporting functions for node module
* running main function when called directly (from node)
* Gilbert2D and Gilbert3D return array of points now (was printing)

zzyzek committed:
updates

* starting 'scan method' for results/experiments
* fattened up the 2d gilbert curve examples

zzyzek committed:
staring pathological figures

zzyzek committed:
fiddling with coherence

* output curve is deflated from fig. 3

zzyzek committed:
simple plotting

* space filling curve reference implementation of hilbert, peano
  and morton curves (2d?) for simple distance binning plots

zzyzek committed:
plots for hilbert curve

zzyzek committed:
hilbert3d curve

* for experiments/plots section

zzyzek committed:
wip

* working on plot scripts

zzyzek committed:
working on experiments and results plots (#8)

* changed norm to 2 (from 1) for point distance
* fixed bug in `create-raw-plots` script (hilbert3 not hilbert3d)
* symlink to gilbert3dpp.js

zzyzek committed:
notes

Jaffers post on SFC

zzyzek committed:
plot updates

* sfc now output the cumulative distribution function (cdf) of
  curve distance vs. embedded distance
* sfjs uses default exponent of 1.5 to accomodate

zzyzek committed:
fiddling with plot creation scripts

zzyzek committed:
automatic scripts

* create "raw" plots of cdf distance binning
* do the data collapse

zzyzek committed:
command line options for data scripts

* initial version of 2d plot (fig_2d.plot)

zzyzek committed:
hilbert3d matches up to gilbert3d

zzyzek committed:
rough draft of data collapse data set figures

* they're pretty rough and I'm using gnuplot directly.
  hopefully they're a good starting point

zzyzek committed:
updates

* removing extraneous sfc.js (plot/sfc.js is the most current version)
* updating sfc.js to do proper data binning
* working on section 4 "experiments and results" in paper

zzyzek committed:
adapt method parameter

* lightly tested for 2d
* untested for 3d
* variable parameter reminiscent of c-style define parameters
  - 0:harmony (jakub width/height/depth choices)
  - 1:hamiltonian (prioritize no notch)
  - 2:axis (force specified axis order, aka 'no-adapt')

zzyzek committed:
experiments and results plots and writing

* might resolve #8

zzyzek committed:
removing function cruft

zzyzek commented:
Initial draft of "Experiments and Results" section (Section 4).

* fixed bug for data binning technique, should do proper averaging of embedded distances
* create data collapse plots
* initial draft of Section 4 written

zzyzek committed:
rough draft

* emphasis on the rough

zzyzek commented:
First completed rough draft (emphasis on the rough).

All sections are written. At the very least need review and most will probably need editing.

There's still some items to do but I'll either create issues for them or update pre-existing issues.

This PR should mostly resolve #8 and #9.

zzyzek committed:
3d adaptive example (3,5,4)

drupol commented:
Nice work!

Do you have a pdf of the current state available?

So sad you haven't used Typst (https://typst.app)

zzyzek committed:
wip

* figure for eccentric split in appendix

zzyzek commented:
The compiled PDF is part of the PR. Here's a [PDF snapshot](https://github.com/jakubcerveny/gilbert-paper/blob/28cd02b523a3f54112f5a028413ee1dd33e131a2/doc/Gilbert.pdf).

zzyzek committed:
wip

* pathological cases (wip)
* eccentric splits (enlarged top row cuboids)
* updated with most recent gilbert3dpp.js copy

zzyzek committed:
figures, extra

* s1 eccentric appendix graphic
* h-split (in case we need it, very rough)
* adding 2d defect reduction argument

zzyzek committed:
pathological figures

* other minor tweaks

zzyzek committed:
appendix figures

zzyzek committed:
eccentric split s1 figure

zzyzek committed:
appendix s2 figures

* added |alpha| << |beta| ~ |gamma|

zzyzek committed:
s0 appendix fig

zzyzek committed:
2d eccentric split figure (appendix)

zzyzek committed:
editing (abstract, intro, related work)

zzyzek committed:
editing

* algorithms, results, conclusion
* working on appendix

zzyzek committed:
editing

zzyzek commented:
This PR is mostly to bring the main paper repo in sync with the latest version.

I think the paper is pretty close to completion (subject to any major course correction or edits). The two major outstanding things I see are the continuity/stability experiments (or deciding if we want them in the first place) and cleaning up some arguments for the eccentric splits in the appendix.

zzyzek committed:
cleaning up arguments for appendix eccentric splits

* only have done S0 and S1

zzyzek committed:
wip

* worked out cases for eccentric splits in appendix
  - still a little scattered, need to take out some extraneous section
  - make all subsubsections into proper sections
  - take out sections
  - provide graphics for each eccentric split subsection
  - put some more descriptive verbage at the top instead of nestling it in
    one of the descriptions

zzyzek committed:
individual eccentric figres

* gamma >>
* beta <<

zzyzek committed:
additional eccentric figures

* alpha <<
* beta <<

zzyzek committed:
all eccentric figures for appendix now produced

zzyzek committed:
eccentric gamma small figures (two.js)

zzyzek committed:
spelling, latest

zzyzek committed:
g-subdivision

zzyzek committed:
updated gilbert3dpp.js and fixed gilbert3d_examples figure

* gilbert3dpp.js has `adapt_method` parameter
* gilbert3d_examples figure now uses parameter
* gilbert3dpp.js has experimental sfc with arbitrary corner endpoints
  and generalized Moore curve

zzyzek committed:
Guiseppe -> Giuseppe

zzyzek committed:
wip

zzyzek committed:
wip

zzyzek committed:
wip

zzyzek committed:
wip

* peony fig
* working on milfoil fig
* updating gilbert3dpp.js with version

zzyzek committed:
block diagrams for hibiscus, peony, milfoil

zzyzek committed:
curve examples peony, hibiscus, milfoil

zzyzek committed:
nicer curve rendering

zzyzek committed:
variants and hellebore (wip)

zzyzek committed:
working on hellebore curve viz

zzyzek committed:
hellebore figure (still rough)

zzyzek committed:
endpoint fix

zzyzek commented:
3d *Hibiscus*, *Peony*, *Milfoil* and *Hellebore* JS curve implementations along with figures (see #13).

zzyzek committed:
checking in

zzyzek committed:
orig gilbert3d examples

zzyzek committed:
wip

* checking in what I have

zzyzek committed:
wip

* still very rough and incomplete

zzyzek committed:
still very rough

zzyzek committed:
wip

zzyzek committed:
wip

* todo
* updates to paper

zzyzek committed:
wip

* updating text
  - much better description
  - needs some more passes to clean up and edit
* still needs eccentric cases
* still needs algorithm placement
* still needs refereces
* see todo

zzyzek committed:
example 2d subdivision

* main sections highlighted and labelled

zzyzek committed:
symlink to pdf image

zzyzek committed:
example 2d eccentric split

zzyzek committed:
html/js code for figures

zzyzek committed:
3d alpha eccentric (simple)

zzyzek committed:
graphics 3d alpha eccentric (simple)

zzyzek committed:
more figures

zzyzek committed:
eccentric 3d figures and examples

zzyzek committed:
wip

zzyzek committed:
2d examples

* added start/end and connection point highlighting

zzyzek committed:
3d main subdivision example

zzyzek committed:
bulk template

zzyzek committed:
removing cruft

* removing unused tex files
* running spell-checker

zzyzek committed:
wip

* adding flat parity graphic
* added parity section back in
* fiddling with alignment/formatting

zzyzek committed:
getting close to a rough draft

* bug fix in 3d algorithm
* spacing for side-by-side figures in algorithm
* editing, writing
* spell checking
* no bib creation in cmp.sh

zzyzek committed:
wip

zzyzek committed:
bug fix in gilbert2d, editing, wip

zzyzek committed:
rough draft

zzyzek committed:
spelling

zzyzek committed:
hilbert ref link fix

zzyzek committed:
bug fix in gilbert3d

* needed else ifs, not just ifs because of yields
